static/js/subs.js

```diff
@@ -726,24 +744,75 @@ function ajaxSubscribe(stream, color) {
                                   $(".stream_change_property_info"));
             }
             // The rest of the work is done via the subscribe event we will get
+
+            // Remove the spinner(if added), and replace with a tick. */
+            if (row_object !== undefined) {
+                /* Re-enable the button to handle requests. */
+                $(row_object).addClass("sub_unsub_button");
+
+                tick.removeClass("hide");
+                loading.destroy_indicator(spinner);
+            }
         },
         error: function (xhr) {
+            if (row_object !== undefined) {
+                /* Re-enable the button to handle requests. */
+                $(row_object).addClass("sub_unsub_button");
+
+                tick.removeClass("hide");
+                loading.destroy_indicator(spinner);
+            }
             ui_report.error(i18n.t("Error adding subscription"), xhr,
                             $(".stream_change_property_info"));
         },
     });
 }
 
-function ajaxUnsubscribe(sub) {
+function ajaxUnsubscribe(sub, row_object) {
     // TODO: use stream_id when backend supports it
+    let tick;
+    let spinner;
+
+    /* If row_object is defined, remove the corresponding tick and replace by
+       a spinner, till the operation completes. */
+    if (row_object !== undefined) {
+        /* Prevent sending multiple requests by removing the button class. */
+        $(row_object).removeClass("sub_unsub_button");
+
+        /* Hide the tick. */
+        tick = $(row_object).find("svg");
+        tick.addClass("hide");
+
+        /* Add a spinner to show the request is in process. */
+        spinner = $(row_object).find("#sub_unsub_status").expectOne();
+        spinner.fadeTo(0, 1);
```

Comment:
Is fading in the spinner the right way to do this?  I would expect to switching more immediately to be normal.

Reply:
In any case, I think this block would be best extracted as a separate function, `display_subscribe_toggle_spinner`.

Reply:
Done.